paddle/framework/shape_inference.h

```diff
@@ -0,0 +1,49 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#pragma once
+
+#include "paddle/framework/ddim.h"
+
+namespace paddle {
+namespace framework {
+
+class InferShapeContextBase;
+
+using ShapeInferenceFn =
+    std::function<void(const framework::InferShapeContextBase& ctx)>;
```

Comment:
it's better to define with `const framework::InferShapeContextBase* ctx` ? we will modify the content explicitly.

Reply:
`const T*` is not fit google C++ style. Only `const T&` or `T*` should be used.

Reply:
done